src/content/subgames.cpp

```diff
@@ -283,35 +298,36 @@ std::vector<WorldSpec> getAvailableWorlds()
 	return worlds;
 }
 
+// bit of a hack but it's the simplest way to do this
+#define SETUP_SETTING(name) \
+	conf.set(#name, conf.get(#name))
+
 bool loadGameConfAndInitWorld(const std::string &path, const SubgameSpec &gamespec)
 {
-	// Override defaults with those provided by the game.
-	// We clear and reload the defaults because the defaults
-	// might have been overridden by other subgame config
-	// files that were loaded before.
-	g_settings->clearDefaults();
-	set_default_settings(g_settings);
-	Settings game_defaults;
-	getGameMinetestConfig(gamespec.path, game_defaults);
-	override_default_settings(g_settings, &game_defaults);
-
 	infostream << "Initializing world at " << path << std::endl;
 
 	fs::CreateAllDirs(path);
 
+	Settings conf;
+	set_default_settings(&conf);
+	override_default_settings(&conf, g_settings);
+	getGameMinetestConfig(gamespec.path, conf);
+
 	// Create world.mt if does not already exist
 	std::string worldmt_path = path + DIR_DELIM "world.mt";
 	if (!fs::PathExists(worldmt_path)) {
-		Settings conf;
-
 		conf.set("gameid", gamespec.id);
-		conf.set("backend", "sqlite3");
-		conf.set("player_backend", "sqlite3");
-		conf.setBool("creative_mode", g_settings->getBool("creative_mode"));
-		conf.setBool("enable_damage", g_settings->getBool("enable_damage"));
+
+		SETUP_SETTING(backend);
```

Comment:
if the key isn't in conf, what value will this take?

Reply:
default